src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1429,6 +1429,58 @@ class SelectStatementSegment(ansi.SelectStatementSegment):
     )
 
 
+class LimitClauseSegment(BaseSegment):
+    """A `LIMIT` clause like in `SELECT`."""
+
+    type = "limit_clause"
+    match_grammar: Matchable = Sequence(
+        "LIMIT",
+        Indent,
+        OneOf(
+            OptionallyBracketed(
+                Sequence(
+                    # Allow a number by itself OR
+                    Ref("NumericLiteralSegment"),
+                    # Arithmetic operations followed by a number any number of times
+                    AnyNumberOf(
+                        Sequence(
+                            Ref("ArithmeticBinaryOperatorGrammar"),
+                            Ref("NumericLiteralSegment"),
+                        ),
+                        optional=True,
+                    ),
+                )
```

Comment:
I think if we're going to extend this we should do something more generic. This also doesn't cover all the possibilities.

Keeping the `Ref("NumericLiteralSegment")` though is still sensible because that's a much faster parse option.

Otherwise I think is should be:
```suggestion
                OneOf(
                    # Allow a number by itself OR
                    Ref("NumericLiteralSegment"),
                    # An arbitrary expression
                    Ref("ExpressionSegment"),
                )
```

Reply:
Thank you @alanmcruickshank! I'll get this sorted and come back to you